client/my-sites/plan-features-2023-grid/components/billing-timeframe.tsx

```diff
@@ -88,12 +98,91 @@ function usePerMonthDescription( {
 	return null;
 }
 
+interface WooExpressMonthlyPromotionProps {
+	billingTimeframe: TranslateResult;
+	monthlyPlanSlug: string;
+	yearlyPlanSlug: string;
+}
+
+const DiscountPromotion = styled.div`
+	text-transform: uppercase;
+	font-weight: 600;
+	color: #306e27;
+	font-size: $font-body-extra-small;
+	margin-top: 6px;
+`;
+
+const WooExpressMonthlyPromotion: FunctionComponent< WooExpressMonthlyPromotionProps > = (
+	props
+) => {
+	const translate = useTranslate();
+	const { billingTimeframe, yearlyPlanSlug, monthlyPlanSlug } = props;
+
+	const annualPlan = getPlans()[ yearlyPlanSlug ];
+	const monthlyPlan = getPlans()[ monthlyPlanSlug ];
+	const planPrices = useSelector( ( state ) => ( {
+		annualPlanMonthlyPrice: getPlanRawPrice( state, annualPlan.getProductId(), true ) || 0,
+		monthlyPlanPrice: getPlanRawPrice( state, monthlyPlan.getProductId() ) || 0,
+	} ) );
+
+	const percentageSavings = Math.round(
+		( 1 - planPrices.annualPlanMonthlyPrice / planPrices.monthlyPlanPrice ) * 100
+	);
+	const discountDescription = percentageSavings
+		? translate( `Save %(percentageSavings)s%% by paying annually`, {
+				args: {
+					percentageSavings,
+				},
+				comment: 'Translators: percentageSavings is the percentage saved by paying annually',
+		  } )
+		: null;
+	return (
+		<div>
+			<div>{ billingTimeframe }</div>
+			<DiscountPromotion>{ discountDescription }</DiscountPromotion>
```

Comment:
The `DiscountPromotion` styles look like the only thing that's new here. The rest are already computed for monthly plans [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-2023-grid/components/billing-timeframe.tsx#L45-L63), right? 

Is there anything outside of the missing plan slugs that require repeating logic for computing the monthly-yearly discount? The whole lot should have been one or two conditions to existing logic as far as I can tell.